### DISCUSSION TITLE
Dynamic Route Error in Next.js when using @payloadcms/next RootLayout

### DISCUSSION THREAD

**Discussion by ab-h-i-n** [tauri-apps/tauri#11762] [Q&A] [answered]
I encountered an issue while using the @payloadcms/next package in my Next.js application. During the build process, the following error is raised:

⨯ Internal error: Error: Route / with dynamic = "error" couldn't be rendered statically because it used headers. See more info here: https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic#dynamic-rendering
at headers (./node_modules/next/dist/server/request/headers.js:39:19)
at RootLayout (./node_modules/@payloadcms/next/dist/layouts/Root/index.js:109:83)
digest: "3971410451"

**Comment by FabianLars** [comment, marked as answer]
Do you know whether payload supports nextjs' static exports? Or does it require a nextjs server at runtime too?

Tauri nextjs apps by default require the static export because tauri doesn't have a nodejs runtime. If static exports don't work you either have to host your frontend on an actual server, or [sidecar](https://v2.tauri.app/develop/sidecar/) it.

**Reply by ab-h-i-n** [reply]
Thankyou for your sugession
